### DISCUSSION TITLE
useSWRInfinite uses stale data after mutation

### DISCUSSION THREAD

**Discussion by 0xc14m1z** [vercel/swr#2494] [Q&A]
# Bug report

## Description / Observed Behavior

We are using `useSWRInfinite` to paginate data by a cursor that we receive from the backend to build kind of a social feed.

When a user posts, we would like to perform an optimistic update such that the new element is prepended to the first page.

To achieve that, we use the bound `mutate` we get from `useSWRInfinite` to prepend the new element in the first page, and update the cursor of the last page, pushing the next selection forward by 1, to skip the new element.

After prepending the new element, the feed, the internal representation and the cache contains correct data.
When fetching additional data, though, it still uses stale data when calling the fetcher.

## Expected Behavior

I'd expect that, when generating the cache key for a new page fetch, the most updated element from the cache is available as `previousPageData`.

## Repro Steps / Code Example

Here is a CodeSandbox with a bit more explained: https://codesandbox.io/s/swr-infinite-mutate-89swnb?file=/src/App.tsx

## Additional Context

Tried with both `swr` version 2.0.3 and 2.0.4

**Comment by piecyk** [comment]
Overall it's becase useSWRInfinite in the end use two caches, one global based on fist page key, and then each page has it’s own, initial run paged one is used.

@0xc14m1z imho current workaround would be to update both, something like https://codesandbox.io/s/swr-infinite-mutate-forked-7hd4nw?file=/src/App.tsx
